src/spi.rs

```diff
@@ -1051,21 +1051,32 @@ where
     }
 
     fn send(&mut self, byte: u8) -> nb::Result<(), Error> {
-        let sr = self.spi.sr.read();
+        // Read from DR register to clear OVR
+        let _ = self.spi.dr.read();
+
+        // Read from SR to clear CRCERR and OVR
+        let mut txe = false;
+        self.spi.sr.modify(|r, w| {
+            txe = r.txe().bit_is_set();
+
+            if r.crcerr().bit_is_set() {
+                w.crcerr().clear_bit();
+            }
+            w
+        });
```

Comment:
Right now, this just blindly clears any errors and doesn't indicate that they occurred in the first place. We still need to indicate that the errors happened after clearing them - it's up to the user to determine what to do after the error is detected.

Reply:
ACK. Unfortunately for that to be useful we need proper generic error types which are still stuck in limbo.